### DISCUSSION TITLE
Why aren't the rate limits editable since it's self-hosted? What telemetric data is sent to AppWrite?

### DISCUSSION THREAD

**Discussion by DankMemeGuy** [appwrite/appwrite#4904] [General]
Since AppWrite is self-hosted, why are[ the rate limits ](https://appwrite.io/docs/rate-limits) imposed hardcoded? Why can't they be changed in the config? Every server is different, and every service is different. If I'm running a web server with 256 GB of RAM, very powerful CPU, NVME SSD, and a secured user system that would discourage abuse, I don't see why this is a hardcoded limit.

Furthermore, there is a statement on that page that says:
`To protect the quality of service from AppWrite, additional rate limits may apply to some actions.`

What does that mean? "Quality of service from AppWrite"? Is there data being sent to AppWrite? What data is this? I haven't seen any mention in the docs anywhere of telemetry data.

**Comment by DankMemeGuy** [comment]
any comment from the appwrite devs?

**Reply by stnguyen90** [reply]
Answered in https://github.com/appwrite/appwrite/issues/4965#issuecomment-1377014247

**Reply by christyjacob4** [reply]
@DankMemeGuy you can also find detailed information on https://appwrite.io/policy/privacy

**Reply by DankMemeGuy** [reply]
> @DankMemeGuy you can also find detailed information on https://appwrite.io/policy/privacy

@christyjacob4 
I'm not sure that answers the question. That Privacy policy sounds like it's more for the actual appwrite.io website, and not the actual AppWrite software.

"Personally identifiable information may include, **but is not limited to**"
"This Usage Data may include information such as your computer's Internet Protocol address (e.g. IP address), browser type, browser version, the pages of our Service that you visit, the time and date of your visit, the time spent on those pages, unique device identifiers and other diagnostic data."

and the list can go on...

There should honestly be two privacy policies. One for the website, and one for the software. Based on that policy, if it applies to the software, then AppWrite is siphoning essentially every piece if data from the AppWrite software and sending it somewhere, including but not limited to sensitive data such as:

1. Email address
2. First name and last name
3. Phone number
4. Address, State, Province, ZIP/Postal code, City
5. Cookies and Usage Data

Notwithstanding the session cookies you are injecting in the software, and everything else listed in that privacy policy.

**Reply by DankMemeGuy** [reply]
@eldadfux @christyjacob4

**Comment by eldadfux** [comment]
The rate limits were originally introduced with the Appwrite Cloud in mind. We do have the intention to shift the to a configurable format.

**Reply by ghost** [reply]
just wondering if there was any progress or roadmap with this?

**Comment by Teraesa** [comment]
Hello, 
Who is this ? and why am i on this email thread??
Regards,
Teraesa Metivier




 

    On Friday, January 13, 2023 at 08:07:26 AM EST, DankMemeGuy ***@***.***> wrote:  
 
 


@eldadfux @christyjacob4

**Reply by DankMemeGuy** [reply]
You're probably subscribed to the thread and receive notifications to comments. Just unsubscribe to the thread.
